api/src/main/java/jakarta/data/metamodel/expression/TextFunctionExpression.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.expression;
+
+import jakarta.data.metamodel.TextExpression;
+
+import java.util.List;
+
+public interface TextFunctionExpression<T>
+        extends FunctionExpression<T,String>, TextExpression<T> {
+
+    String CONCAT = "concat";
+    String UPPER = "upper";
+    String LOWER = "lower";
+    String LEFT = "left";
+    String RIGHT = "right";
+
+    static <T> TextFunctionExpression<T> of(String name, TextExpression<T> argument) {
```

Comment:
I would go with an enum with the function that we support.

Otherwise, I could put:

```java
   return TextFunctionExpression.of("gavin", string, this);
  
 ``` 
 
 Unless, I am missing somepoint

Reply:
@otaviojava Remember that:

- JPQL defines extra functions that don't exist in Jakarta Data, and
- actual datastore have their own functions, even user-written functions.

so as I said above, eventually it's *extremely* likely that we'll want to let people call functions which aren't required by the spec.